src/nnpack/libnnpack.jl

```diff
@@ -0,0 +1,145 @@
+end
+
+function nnp_fully_connected_output(x::AbstractArray{Float32,2}, w::AbstractArray{Float32,2}, y::AbstractArray{Float32,2}; profile = nothing, threadpool = nothing)
+    profile = profile == nothing ? nnp_profile() : profile
+    threadpool = threadpool === nothing ? pthreadpool_create() : threadpool
+    nnp_fully_connected_output(size(x, 2), size(x, 1), size(w, 1), x, w, y, threadpool, profile)
+    y
+end
+
+function nnp_fully_connected_inference_f16f32(input_channels, output_channels, input, kernel, output, threadpool)
+    @check ccall((:nnp_fully_connected_inference_f16f32, libnnpack), nnp_status, (Csize_t, Csize_t, Ptr{Cfloat}, Ptr{Cvoid}, Ptr{Cfloat}, pthreadpool_t), input_channels, output_channels, input, kernel, output, threadpool)
+end
+
+nnp_fully_connected_inference_f16f32(x::AbstractVector{Float32}, w::AbstractArray{Float16,2}, y::AbstractVector{Float32}; threadpool = nothing) =
+    nnp_fully_connected_inference(reshape(x, size(x), 1), w, reshape(y, size(y), 1), threadpool = threadpool)
+
+function nnp_fully_connected_inference_f16f32(x::AbstractMatrix{Float32}, w::AbstractArray{Float16,2}, y::AbstractMatrix{Float32}; threadpool = nothing)
+    threadpool = threadpool === nothing ? pthreadpool_create() : threadpool
+    nnp_fully_connected_inference(size(x, 1), size(y, 1), x, w, y, threadpool)
+    y
+end
+
+function nnp_fully_connected_inference(input_channels, output_channels, input, kernel, output, threadpool)
+    @check ccall((:nnp_fully_connected_inference, libnnpack), nnp_status, (Csize_t, Csize_t, Ptr{Cfloat}, Ptr{Cfloat}, Ptr{Cfloat}, pthreadpool_t), input_channels, output_channels, input, kernel, output, threadpool)
+end
+
+nnp_fully_connected_inference(x::AbstractVector{Float32}, w::AbstractArray{Float32,2}; threadpool = nothing) =
+    nnp_fully_connected_inference(reshape(x, size(x), 1), w, threadpool = threadpool)
+
+function nnp_fully_connected_inference(x::AbstractMatrix{Float32}, w::AbstractMatrix{Float32}, y::AbstractMatrix{Float32}; threadpool = nothing)
+    threadpool = threadpool === nothing ? pthreadpool_create() : threadpool
+    nnp_fully_connected_inference(size(x, 1), size(y, 1), x, w, y, threadpool)
+    y
+end
+
+function nnp_max_pooling_output(batch_size, channels, input_size, input_padding, pooling_size, pooling_stride, input, output, threadpool)
+    @check ccall((:nnp_max_pooling_output, libnnpack), nnp_status, (Csize_t, Csize_t, nnp_size, nnp_padding, nnp_size, nnp_size, Ptr{Cfloat}, Ptr{Cfloat}, pthreadpool_t), batch_size, channels, input_size, input_padding, pooling_size, pooling_stride, input, output, threadpool)
+end
+
+function nnp_max_pooling_output(x::AbstractArray{Float32,4}, y::AbstractArray{Float32,4}, kernel::Tuple; padding = 0, stride = 1, threadpool = nothing)
```

Comment:
Shouldn't these functions all use the shared threadpool now?
```suggestion
function nnp_max_pooling_output(x::AbstractArray{Float32,4}, y::AbstractArray{Float32,4}, kernel::Tuple; padding = 0, stride = 1, threadpool = shared_threadpool[])
```